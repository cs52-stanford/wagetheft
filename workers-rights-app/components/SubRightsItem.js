import React from "react";
import { View, Text, StyleSheet, TouchableOpacity, Image } from "react-native";
import PropTypes from "prop-types";
import Colors from "../constants/Colors";

/*
 * Functional Component Defintion: SubRightsItem
 *
 * This component outlines the look of SubRightItem
 * on the SubRightsScreen.
 *
 */

const SubRightsItem = ({ onSelect, img, title }) => {
  return (
    <TouchableOpacity
      onPress={onSelect}
      style={styles.subrights}
      testID="subItem"
    >
      <View style={styles.txtContainer}>
        <Image style={styles.icon} source={img} />
        <View style={styles.contentContainer}>
          <Text style={styles.subrightstxt} testID="text">
            {title}
          </Text>
        </View>
        <Text style={styles.subrightstxt}>â†’</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  subrights: {
    alignItems: "center",
    padding: 15,
    margin: 7,
    borderRadius: 8,
    shadowOpacity: 0.25,
    shadowOffset: { width: 2, height: 5 },
    elevation: 3,
    backgroundColor: Colors.lightOrange,
  },
  contentContainer: {
    maxWidth: "80%",
  },
  subrightstxt: {
    fontSize: 16,
    paddingHorizontal: 2,
    textAlign: "center",
  },
  txtContainer: {
    flexDirection: "row",
    width: "100%",
    justifyContent: "space-between",
    alignItems: "center",
    flex: 1,
  },
  icon: {
    width: 30,
    height: 30,
  },
});

SubRightsItem.propTypes = {
  onSelect: PropTypes.func.isRequired,
  img: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

export default SubRightsItem;
