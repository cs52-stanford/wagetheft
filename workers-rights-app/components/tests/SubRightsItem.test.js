import React from "react";
import SubRightsItem from "../SubRightsItem";
import { render, fireEvent } from "@testing-library/react-native";

describe("<SubRightsItem /> Tests", () => {
  it("should match snapshot", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <SubRightsItem title="Test Title" onSelect={mockPressFunction} img={1} />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it("should display text properly", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <SubRightsItem title="Test Title" onSelect={mockPressFunction} img={1} />
    );
    const textComponent = rendered.getByTestId("text");
    expect(textComponent.props.children).toEqual("Test Title");
  });

  it("should fire onPress events", () => {
    const onSelect = jest.fn();
    const rendered = render(
      <SubRightsItem title="Test Title" onSelect={onSelect} img={1} />
    );
    const buttonComponent = rendered.getByTestId("subItem");
    fireEvent(buttonComponent, "press");
    expect(onSelect).toHaveBeenCalled();
  });
});
