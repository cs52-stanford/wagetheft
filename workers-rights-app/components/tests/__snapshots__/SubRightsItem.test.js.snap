// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubRightsItem /> Tests should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF6EB",
      "borderRadius": 8,
      "elevation": 3,
      "margin": 7,
      "opacity": 1,
      "padding": 15,
      "shadowOffset": Object {
        "height": 5,
        "width": 2,
      },
      "shadowOpacity": 0.25,
    }
  }
  testID="subItem"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <View
      style={
        Object {
          "maxWidth": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingHorizontal": 2,
            "textAlign": "center",
          }
        }
        testID="text"
      >
        Test Title
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingHorizontal": 2,
          "textAlign": "center",
        }
      }
    >
      â†’
    </Text>
  </View>
</View>
`;
